src/handlers/mod.rs

```diff
@@ -397,9 +371,10 @@ pub fn upload_blob_prechecks(
     // Only check validity if the header is present.
     // This is safe, because in the PUT /upload implementation the actual blob size is checked.
     if let Some(content_length) = content_length {
-        if let Ok(content_length) = content_length.parse::<u64>() {
-            let blob_size_in_mb = bytes_to_mb(content_length as f64);
-            if blob_size_in_mb > app_state.config.upload.max_size {
+        if let Ok(content_length) = content_length.parse::<f64>() {
```

Comment:
Oops i only notice this now. This isn't strictly correct. The `Content-Length` header should be parsed as an unsigned integer, not a floating point number. Better to keep it as `<u64>` then cast it to f64 later.

Reply:
Oh right, I'll make a new PR for this